air-quality-backend/system_tests/forecast_etl_suite/cams_known_grib_test.py

```diff
@@ -0,0 +1,216 @@
+            "forecast_base_time": datetime(2024, 6, 4, 0, 0, 0, tzinfo=timezone.utc),
+            "location_type": "city",
+            "source": "cams-production",
+        },
+        {
+            "name": "Vancouver",
+            "forecast_valid_time": datetime(2024, 6, 4, 3, 0, 0),
+            "forecast_base_time": datetime(2024, 6, 4, 0, 0, 0),
+            "no2_value": 10.115771485462055,
+            "no2_aqi_level": 1,
+            "o3_value": 63.39522524659603,
+            "o3_aqi_level": 2,
+            "so2_value": 2.000387638160907,
+            "so2_aqi_level": 1,
+            "pm10_value": 5.810208243195257,
+            "pm10_aqi_level": 1,
+            "pm2_5_value": 3.6327574082026914,
+            "pm2_5_aqi_level": 1,
+        },
+    ),
+    # Extreme Longitude Check 180
+    (
+        {
+            "name": "Wellington",
+            "forecast_valid_time": datetime(2024, 6, 4, 0, 0, 0, tzinfo=timezone.utc),
+            "forecast_base_time": datetime(2024, 6, 4, 0, 0, 0, tzinfo=timezone.utc),
+            "location_type": "city",
+            "source": "cams-production",
+        },
+        {
+            "name": "Wellington",
+            "forecast_valid_time": datetime(2024, 6, 4, 0, 0, 0),
+            "forecast_base_time": datetime(2024, 6, 4, 0, 0, 0),
+            "no2_value": 0.4782831135385457,
+            "no2_aqi_level": 1,
+            "o3_value": 56.25565199915405,
+            "o3_aqi_level": 2,
+            "so2_value": 0.3777866472717213,
+            "so2_aqi_level": 1,
+            "pm10_value": 16.71105682154486,
+            "pm10_aqi_level": 1,
+            "pm2_5_value": 9.073147098209494,
+            "pm2_5_aqi_level": 1,
+        },
+    ),
+    # Close to 0 longitude
+    (
+        {
+            "name": "London",
+            "forecast_valid_time": datetime(2024, 6, 4, 6, 0, 0, tzinfo=timezone.utc),
+            "forecast_base_time": datetime(2024, 6, 4, 0, 0, 0, tzinfo=timezone.utc),
+            "location_type": "city",
+            "source": "cams-production",
+        },
+        {
+            "name": "London",
+            "forecast_valid_time": datetime(2024, 6, 4, 6, 0, 0),
+            "forecast_base_time": datetime(2024, 6, 4, 0, 0, 0),
+            "no2_value": 19.136068918675015,
+            "no2_aqi_level": 1,
+            "o3_value": 9.499161050717573,
+            "o3_aqi_level": 1,
+            "so2_value": 3.8678773476028074,
+            "so2_aqi_level": 1,
+            "pm10_value": 24.70699837214256,
+            "pm10_aqi_level": 2,
+            "pm2_5_value": 17.1565041916872,
+            "pm2_5_aqi_level": 2,
+        },
+    ),
+]
+
+
+@mock.patch.dict(
+    os.environ,
+    {"FORECAST_BASE_TIME": "2024-6-4 00"},
+)
+@pytest.mark.parametrize("query_params, expected_values", test_cases)
+def test__extreme_longitudes__to_assert_exact_values(query_params, expected_values):
+    delete_database_data("forecast_data")
+    main()
```

Comment:
If I understand this file correctly, all the tests rely on these two steps to have run.

This means that if this test runs after another one the first would fail.

It's usually best practice to make tests independent of each other.

In pytest you can do this with fixtures e.g:
```python
@pytest.fixture
def setup_data():
    # Set up code
    delete_database_data("forecast_data")
    main()
    yield
    # Tear down code

def test_one(setup_data):
    # Test code for test_one
    assert ...

def test_two(setup_data):
    # Test code for test_two
    assert ...
```

If you really need order to be guaranteed we can probably find a way but I don't think you do here

Reply:
Nice this is useful, I tried using fixtures before. Now I see that the issue was that I didn't then add that fixture as a parameter. Does this mean that we are deleting and repopulating the database each test?